### PR TITLE
Keyword parameters

### LINKED ISSUE #20: Support Keyword Args
Some of the functionality released in 2.0.0 uses positional arguments, which are fine in some circumstances, but since a lot of these functions have optional parameters, keyword arguments provide a more readable and useable interface.

We will need to continue to support the positional arguments in 2.x, but we should update the methods and docs to prefer keyword arguments. I think these methods can be tweaked, using defaults and keywords after positionals, plus some presence checking to support the existing positional args, but prefer keyword args.

Consumer methods can be positional since they are a single, required parameter (and more of an interface than a callable method).

### PR DESCRIPTION
**Pull request by lpradovera**
Fixes #20

### CODE REVIEW

**Comment by lpradovera** [comment]
@bryanrite this seems to be working quite well, and incidentally supports mixed parameters (though I would not recommend anyone doing that).

**Comment by lpradovera** [comment]
@bryanrite this might be GTG, please take a look.

File: examples/relay/outbound_consumer.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@ def ready
     logger.info 'Dialing out'
     call = client.calling.new_call(from: ENV['FROM_NUMBER'], to: ENV['TO_NUMBER'])
     call.dial
-    call.play_tts 'the quick brown fox jumps over the lazy dog', 'en-US', 'male'
+    call.play_tts sentence: 'the quick brown fox jumps over the lazy dog', 'en-US', 'male'
```
Comment (bryanrite) [inline examples/relay/outbound_consumer.rb]: Example should use all keyword args.

File: lib/signalwire/relay/calling/call_convenience_methods.rb [hunk 2]
```diff
@@ -16,35 +16,49 @@ def play_silence!(duration)
       play! silence_payload(duration)
     end
 
-    def play_tts(sentence, language='en-US', gender='female')
+    def play_tts(sentence_p=nil, language_p=Relay::DEFAULT_LANGUAGE, gender_p=Relay::DEFAULT_GENDER, sentence: nil, language: Relay::DEFAULT_LANGUAGE, gender: Relay::DEFAULT_GENDER)
```
Comment (bryanrite) [inline lib/signalwire/relay/calling/call_convenience_methods.rb]: The other SDKs use `text:` which I think is nicer than `sentence:`.

File: spec/signalwire/relay/calling/call_spec.rb [hunk 3]
```diff
@@ -72,4 +72,101 @@
...
+        it "handles positional parameters" do
+          subject.play_tts(sentence_obj)
+        end
+  
+        it "handles keyword parameters" do
+          subject.play_tts(sentence: sentence_obj)
+        end
+  
+        context "optional parameters" do
+          let(:language) { "it-IT" }
```
Comment (bryanrite) [inline spec/signalwire/relay/calling/call_spec.rb:164]: 🇮🇹

**Review by bryanrite — CHANGES_REQUESTED** [review]
A few small changes to be inline with the other SDKs.

I like the `set_parameters` method, looks great.

**Review by bryanrite — APPROVED** [review]

**Pull request merged**